src/tlo/analysis/utils.py

```diff
@@ -311,19 +311,15 @@ def summarize(results: pd.DataFrame, only_mean: bool = False, collapse_columns:
     Finds mean value and 95% interval across the runs for each draw.
     """
 
-    summary = pd.DataFrame(
-        columns=pd.MultiIndex.from_product(
-            [
-                results.columns.unique(level='draw'),
-                ["mean", "lower", "upper"]
-            ],
-            names=['draw', 'stat']),
-        index=results.index
+    summary = pd.concat({
+        'mean': results.groupby(axis=1, by='draw', sort=False).mean(),
+        'lower': results.groupby(axis=1, by='draw', sort=False).quantile(0.025),
+        'upper': results.groupby(axis=1, by='draw', sort=False).quantile(0.975),
+    },
```

Comment:
```suggestion
    summary = pd.concat(
        {
            'mean': results.groupby(axis=1, by='draw', sort=False).mean(),
            'lower': results.groupby(axis=1, by='draw', sort=False).quantile(0.025),
            'upper': results.groupby(axis=1, by='draw', sort=False).quantile(0.975),
        },
```
Matching indentation of opening and closing delimiters makes it a bit easier to see the grouping of the items here.